atd-vze/src/views/Crashes/Notes.js

```diff
@@ -1,42 +1,95 @@
-import React, { Component } from "react";
-import { useQuery } from "@apollo/react-hooks";
-import { Card, CardHeader, CardBody, CardFooter, Table } from "reactstrap";
+import React, { useState } from "react";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import { Card, CardHeader, CardBody, CardFooter, Table, Input, Button, Col } from "reactstrap";
 import moment from "moment";
 import { notesDataMap } from "./notesDataMap";
-import { GET_NOTES } from "../../queries/notes";
+import { GET_NOTES, INSERT_NOTE } from "../../queries/notes";
+import { useAuth0, isReadOnly } from "../../auth/authContext";
 
 // declare a notes component
 const Notes = ({ crashId }) => {
 
+  // add a state variable to manage value when new note in entered
+  const [newNote, setNewNote] = useState("");
+
+  // disable edit features if only role is "readonly"
+  const { getRoles } = useAuth0();
+  const roles = getRoles();
+
   // fetch data from database using graphQL query
   const { loading, error, data, refetch } = useQuery(GET_NOTES, {
     variables: { crashId },
   });
 
+  // declare insert mutation function
+  const [addNote] = useMutation(INSERT_NOTE);
+
   if (loading) return "Loading...";
   if (error) return `Error! ${error.message}`;
 
   const tableName = "notes";
   const keyField = "id";
   const fieldConfig = notesDataMap[0];
 
+  // function to handle add button click
+  const handleAddNoteClick = () => {
+    const userEmail = localStorage.getItem("hasura_user_email");
+    addNote({
+      variables: {
+        note: newNote,
+        crashId: crashId,
+        userEmail: userEmail
+      }
+    }).then(response => {
+      setNewNote("");
+      refetch();
+    });
```

Comment:
It's not required but it would be a best practice to use `.catch` to watch for any errors in the mutation process. after the `.then()` closes, you could add `).catch( error => console.error(error)`.

This could at least help a dev triage if a bug is reproducible.

More info of error handling in JS:
- https://lucymarmitchell.medium.com/using-then-catch-finally-to-handle-errors-in-javascript-promises-6de92bce3afc
- https://javascript.info/promise-error-handling

Reply:
I don't see your update. Maybe try to push again?

Reply:
![image](https://user-images.githubusercontent.com/67130963/178055552-4ff5aa51-be4f-4e7d-b6c5-60b0a042e5a9.png)

You dont see this?

Reply:
Thank you, I do now. And I'm not sure if that chaining will work as is. Here is what I think you need:

```js
    }).then(response => {
      setNewNote("")
      refetch();
    }).catch(error => console.error(error));
```

As you have it written, the catch would run after `setNewNote()` errors (assuming its a Promise). You want it chained after the `.then()` errors.

Reply:
Oooh okay, I was unsure about the chaining. that makes a lot more sense. thank you mateo!